From e815d50660403c5dc6c54d624dc3ccf30cf9425a Mon Sep 17 00:00:00 2001
From: Ravi Rama <rrama@arista.com>
Date: Wed, 16 Apr 2025 11:18:38 -0700
Subject: [PATCH] dont use CPU GPIOs to control XGBE port LEDs

---
 .../Keith/KTInterface/KTFchInitEnv.c          | 52 +++++++++----------
 1 file changed, 26 insertions(+), 26 deletions(-)

diff --git a/AMD/AmdEmbedded/AgesaModulePkg/Fch/Keith/FchKeithCore/Keith/KTInterface/KTFchInitEnv.c b/AMD/AmdEmbedded/AgesaModulePkg/Fch/Keith/FchKeithCore/Keith/KTInterface/KTFchInitEnv.c
index 6aa9fe9b..a5bb4ead 100644
--- a/AMD/AmdEmbedded/AgesaModulePkg/Fch/Keith/FchKeithCore/Keith/KTInterface/KTFchInitEnv.c
+++ b/AMD/AmdEmbedded/AgesaModulePkg/Fch/Keith/FchKeithCore/Keith/KTInterface/KTFchInitEnv.c
@@ -46,9 +46,9 @@ FchInitEnv (
 {
   AGESA_STATUS        Status;
   //EMBDEV-16636_Start
-  UINT32                 PortsPropertyReg;
-  UINT32                 GpioValue;
-  UINT32                 GpioAddress;
+  //UINT32                 PortsPropertyReg;
+  //UINT32                 GpioValue;
+  //UINT32                 GpioAddress;
   //EMBDEV-16636_End
 
   // Override internal data with IDS (Optional, internal build only)
@@ -58,34 +58,34 @@ FchInitEnv (
   Status = FchTaskLauncher (&FchInitEnvTasksTable[0], FchDataBlock, TpFchInitEnvDispatching);
 
   //EMBDEV-16636_Start
-  GpioAddress = ACPI_MMIO_BASE + GPIO_BANK0_BASE + (PcdGet8(PcdXgbePort0Led1Gpio)*4);
-  RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT23), BIT23); //OutputEnable : 0-Output is disabled on the pin, 1-Output is enabled on the pin
-  RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT22), 0);     //OutputValue  : 0-low, 1-high
+  //GpioAddress = ACPI_MMIO_BASE + GPIO_BANK0_BASE + (PcdGet8(PcdXgbePort0Led1Gpio)*4);
+  //RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT23), BIT23); //OutputEnable : 0-Output is disabled on the pin, 1-Output is enabled on the pin
+  //RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT22), 0);     //OutputValue  : 0-low, 1-high
 
-  GpioAddress = ACPI_MMIO_BASE + GPIO_BANK0_BASE + (PcdGet8(PcdXgbePort0Led2Gpio)*4);
-  RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT23), BIT23);
-  RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT22), 0);
+  //GpioAddress = ACPI_MMIO_BASE + GPIO_BANK0_BASE + (PcdGet8(PcdXgbePort0Led2Gpio)*4);
+  //RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT23), BIT23);
+  //RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT22), 0);
 
-  GpioAddress = ACPI_MMIO_BASE + GPIO_BANK0_BASE + (PcdGet8(PcdXgbePort1Led1Gpio)*4);
-  RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT23), BIT23);
-  RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT22), 0);
+  //GpioAddress = ACPI_MMIO_BASE + GPIO_BANK0_BASE + (PcdGet8(PcdXgbePort1Led1Gpio)*4);
+  //RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT23), BIT23);
+  //RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT22), 0);
 
-  GpioAddress = ACPI_MMIO_BASE + GPIO_BANK0_BASE + (PcdGet8(PcdXgbePort1Led2Gpio)*4);
-  RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT23), BIT23);
-  RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT22), 0);
+  //GpioAddress = ACPI_MMIO_BASE + GPIO_BANK0_BASE + (PcdGet8(PcdXgbePort1Led2Gpio)*4);
+  //RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT23), BIT23);
+  //RwMem (GpioAddress,AccessWidth32, ~(UINT32)(BIT22), 0);
 
   //Writing GPIO numbers to Port Property register
-  GpioValue   = (UINT32) PcdGet8(PcdXgbePort0Led2Gpio); //Default 0x5B(AGPIO091)
-  GpioValue   = (GpioValue << 8) | ((UINT32)PcdGet8(PcdXgbePort0Led1Gpio)); //Default 0x0A(AGPIO010)
-  FchSmnRead (0, FCH_KT_SMN_XGBE_MMIO0_BASE + 0x814, &PortsPropertyReg, NULL);
-  PortsPropertyReg  = (PortsPropertyReg & 0xFFFF0000) | GpioValue;
-  FchSmnWrite (0, FCH_KT_SMN_XGBE_MMIO0_BASE + 0x814, &PortsPropertyReg, NULL);
-
-  GpioValue   = (UINT32) PcdGet8(PcdXgbePort1Led2Gpio);// Default 0x9D(AGPIO157)
-  GpioValue   = (GpioValue << 8) | ((UINT32)PcdGet8(PcdXgbePort1Led1Gpio)); // Default 0x90(AGPIO144)
-  FchSmnRead (0, FCH_KT_SMN_XGBE_MMIO0_BASE + 0x834, &PortsPropertyReg, NULL);
-  PortsPropertyReg  = (PortsPropertyReg & 0xFFFF0000) | GpioValue;
-  FchSmnWrite (0, FCH_KT_SMN_XGBE_MMIO0_BASE + 0x834, &PortsPropertyReg, NULL);
+  //GpioValue   = (UINT32) PcdGet8(PcdXgbePort0Led2Gpio); //Default 0x5B(AGPIO091)
+  //GpioValue   = (GpioValue << 8) | ((UINT32)PcdGet8(PcdXgbePort0Led1Gpio)); //Default 0x0A(AGPIO010)
+  //FchSmnRead (0, FCH_KT_SMN_XGBE_MMIO0_BASE + 0x814, &PortsPropertyReg, NULL);
+  //PortsPropertyReg  = (PortsPropertyReg & 0xFFFF0000) | GpioValue;
+  //FchSmnWrite (0, FCH_KT_SMN_XGBE_MMIO0_BASE + 0x814, &PortsPropertyReg, NULL);
+
+  //GpioValue   = (UINT32) PcdGet8(PcdXgbePort1Led2Gpio);// Default 0x9D(AGPIO157)
+  //GpioValue   = (GpioValue << 8) | ((UINT32)PcdGet8(PcdXgbePort1Led1Gpio)); // Default 0x90(AGPIO144)
+  //FchSmnRead (0, FCH_KT_SMN_XGBE_MMIO0_BASE + 0x834, &PortsPropertyReg, NULL);
+  //PortsPropertyReg  = (PortsPropertyReg & 0xFFFF0000) | GpioValue;
+  //FchSmnWrite (0, FCH_KT_SMN_XGBE_MMIO0_BASE + 0x834, &PortsPropertyReg, NULL);
   //EMBDEV-16636_End
 
   IDS_HDT_CONSOLE (FCH_TRACE, "  FchInitEnv Exit... Status = [0x%x]\n", Status);
-- 
2.47.0

